dialect/sql/builder.go

```diff
@@ -2960,7 +2960,16 @@ type WindowBuilder struct {
 // Using this function will assign a each row a number, from 1 to N, in the
 // order defined by the ORDER BY clause in the window spec.
 func RowNumber() *WindowBuilder {
-	return &WindowBuilder{fn: "ROW_NUMBER"}
+	return WindowSelector(Select("ROW_NUMBER()"))
+}
+
+// WindowSelector returns a new window clause with a custom selector allowing
+// for custom windown functions.
+//
+//	s1 := WindowSelector(SelectExpr(Expr("ROW_NUMBER()")))
+//	s2 := WindowSelector(Select("ROW_NUMBER()"))
+func WindowSelector(s *Selector) *WindowBuilder {
```

Comment:
The first argument/grammar-rule holds the window function, right? So I'm not sure using a `Selector` is right here. 

What do you think about implementing it like predicates? i.e. `P(func(*sql.Builder)`. The reason it gets a function and not a builder is that the function is invoked only on `Query()` call and may use the same dialect-builder.

```go
func RowNumber() *WindowBuilder {
	return Window(func(b *sql.Builder) {
		b.WriteString("ROW_NUMBER")
	})
}
```

We can create helper functions like `RowNumber` in the future if we feel the API is not intuitive.

Reply:
Updated! let me know if it's something you are looking for